src/prost/src/lib.rs

```diff
@@ -289,12 +289,33 @@ impl stream_plan::StreamNode {
 
     /// Find the external stream source info inside the stream node, if any.
     ///
-    /// Returns `source_id`.
-    pub fn find_source_backfill(&self) -> Option<u32> {
+    /// Returns (`source_id`, `upstream_source_fragment_id`, `upstream_actor_id`).
+    ///
+    /// Note: we must get upstream fragment id from the merge node, not from the fragment's
+    /// `upstream_fragment_ids`. e.g., DynamicFilter may have 2 upstream fragments, but only
+    /// one is the upstream source fragment.
+    ///
+    /// Note: `merge.upstream_actor_id` need to be used with caution.
+    /// DO NOT USE if the `StreamNode` is from `Fragment` meta model.
+    /// OK to use if the `StreamNode` is from `TableFragments` proto.
```

Comment:
Why, can you elaborate more? This sounds type-unsafe.

Reply:
updated